Lib/Protocols/IdFTPServer.pas

```diff
@@ -6018,6 +6053,56 @@ function TIdFTPServer.MLSFEATLine(const AFactMask: TIdMLSDAttrs;
   end;
 end;
 
+function ParseCSIDParams(const AParams : String) : string;
+var LBuf,
+    LValueValue : String;
+    LValueName : String;
+    LCltName : String;
+    LCltVersion : String;
+begin
+   Result := '';
+   LBuf := AParams;
+   repeat
+      LValueValue := Fetch(LBuf,';');
+      LValueName := TrimLeft(Fetch(LValueValue,'='));
+      case PosInStrArray(LValueName,['Name','Version']) of
+        0 : LCltName := LValueValue;
+        1 : LCltVersion := LValueValue;
+      end;
+   until LBuf = '';
+   Result := Trim(LCltName + ' ' + LCltVersion);

```

Comment:
I would suggest including the `OS` value to mimic the `PlatformDescription` portion of the `CLNT` command.

Reply:
if you mean, rename the variable OS, I have made that change.

Reply:
No, not rename. I meant if the client sends `OS`/`OSVer` in the `CSID` command then include that info in the string that you are passing to the `OnClientID` event, to mimic what `CommandCLNT` passes to `OnClientID` if that info  is present.

Reply:
I don't think there is a PlatformDescription that is sent to the server with the CLNT command.  FileZilla sends "CLNT FileZilla" and WS-FTP-Professional sends "CLNT WS_FTP_Professional 12.9.1".  And further, the draft https://datatracker.ietf.org/doc/html/draft-peterson-streamlined-ftp-comm  and-extensions-10 does not state that the client can send an "OS" as a parameter to the CSID command.